docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst

```diff
@@ -79,17 +80,24 @@ You can also specify:
 
 - .. include:: ../reusables/threads-query-execution.rst
 
-
-.. pull-quote::
+. pull-quote::
 
    Upgrading databases
 
-   If the CodeQL queries you want to use are newer than the
-   extractor used to create the database, then you may see a message telling you
+   For databases that were created by CodeQL CLI v2.3.3 or earlier, you will need
+   to explicitly upgrade the database before you can run an analysis with a newer
+   version of the CodeQL CLI. If this step is necessary, then you will see a message telling you
    that your database needs to be upgraded when you run ``database analyze``.
-   You can quickly upgrade a database by running the ``database upgrade``
-   command. For more information, see ":doc:`Upgrading CodeQL databases
-   <upgrading-codeql-databases>`."
+
+   For databases that were created by CodeQL CLI v2.3.4 or later, the CLI will implicitly run any
+   required upgrades. Explicitly running the upgrade command is not necessary.
+
+   CodeQL databases are upgraded by running the following command::
+
+       codeql database upgrade <database>
+
+   For full details of all the options you can use when upgrading databases,
+   see the "`database upgrade <../manual/database-upgrade>`__"  reference documentation.
```

Comment:
I think we can probably drop these lines altogether as the first two sentences of this note tell users everything they need to know.

```suggestion
```